Formula/dovecot.rb

```diff
@@ -77,6 +77,15 @@ def install
   def caveats; <<-EOS.undent
     For Dovecot to work, you may need to create a dovecot user
     and group depending on your configuration file options.
+
+    Dovecot by default requires a higher limit on open files per
+    process than the Mac OS X standard of 256 open files. In order
+    to silence the warning from dovecot, the launchd plist sets
+    both hard and soft resource limits of 512 open files. It is
+    suggested you also set default_client_limit = 512 in
+    10-master.conf. If you are running a production server, set
```

Comment:
Thank you for your suggestion @MikeMcQuaid. I looked into it, and could not find a compile flag for dovecot that would compile it with a different default_client_limit.

So I have a few options:

1. Do a string manipulation of Cellar/dovecot/[version]/share/doc/dovecot/example-config/conf.d/10-master.conf adding `default_client_limit = 512` (and a comment) underneath the #default_client_limit=1000 

This would help set up new installations of dovecot, but would not do much for users who already have dovecot installed and are just upgrading. (I don't really want to do a string manipulation on users' already configured conf files for obvious reasons).

2. Add a file to Cellar/dovecot/[version]/share/doc/dovecot/example-config/conf.d/ called, perhaps, `99-homebrew-defaults.conf`, with the default_client_limit set to 512 and a comment describing the why, and hope users see it. 

3. Add the above file to user's existing installs and finding a way to tell them that, if they already had a system configured, this file has been added.

4. Ignore the fact that most users of dovecot on a Mac OS X system are probably running it only for their own use and don't need the default limit set to 1000 (obviously Apple thinks a max files per process limit of 256 is fine), and just set the .plist file to allow 1000 files. This would set higher limits on the processes, which could potentially lead to trouble, but I would not be reconfiguring existing users' configurations 💣 , nor would we need any explanatory note in the caveats.

I think 1 or 4 are the best options, and if we want to strictly minimize the amount of messaging to users, option 4 is probably the way to go, though it does mean the program has higher allowances than necessary for almost all our users.